grakn-graph/src/main/java/ai/grakn/graph/internal/RoleTypeImpl.java

```diff
@@ -51,7 +50,15 @@
  *
  */
 class RoleTypeImpl extends TypeImpl<RoleType, Instance> implements RoleType{
-    RoleTypeImpl(AbstractGraknGraph graknGraph, Vertex v, Optional<RoleType> type, Optional<Boolean> isImplicit) {
+    RoleTypeImpl(AbstractGraknGraph graknGraph, Vertex v) {
+        super(graknGraph, v);
+    }
+
+    RoleTypeImpl(AbstractGraknGraph graknGraph, Vertex v, RoleType type) {
+        super(graknGraph, v, type);
+    }
```

Comment:
If we get rid of the `if` statements above, maybe we can get rid of this constructor.